glibc-2.40.patch

```diff
@@ -0,0 +1,53 @@
+--- a/nptl/allocatestack.c
++++ b/nptl/allocatestack.c
+@@ -210,7 +210,7 @@ advise_stack_range (void *mem, size_t size, uintptr_t pd, size_t guardsize)
+    new stack or reusing a cached stack of sufficient size.
+    ATTR must be non-NULL and point to a valid pthread_attr.
+    PDP must be non-NULL.  */
+-static int
++int
+ allocate_stack (const struct pthread_attr *attr, struct pthread **pdp,
+                void **stack, size_t *stacksize)
+ {
+@@ -403,6 +403,18 @@ allocate_stack (const struct pthread_attr *attr, struct pthread **pdp,
+                }
+            }
+
++  /* Initialize the TCB.  All initializations with zero should be
++     performed in 'get_cached_stack'.  This way we avoid doing this if
++     the stack freshly allocated with 'mmap'.  */
++
++#if TLS_TCB_AT_TP
++    /* Reference to the TCB itself.  */
++    pd->header.self = pd;
++
```

Comment:
These lines are moved from pthread_create(), because fiber_create() doesn't have the definition of *pd, and can not do it outside.